refactor(overview): improve case overview tab handling and labels

alonp99 opened the pull request:
- Update tab labels for clarity and consistency
- Simplify tab entry lookup by using Object.entries
- Enhance risk indicators mapping for improved readability


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Refactor**
  - Updated display labels for better clarity (e.g., "Store Info," "KYC," and "Web Presence").
  - Revised the logic for presenting risk indicator information, including updated naming and sorting.
  - Reorganized risk indicator categories and refreshed indicator text in demo data for improved user experience.
  - Enhanced visual presentation of badges in the user interface.
- **Bug Fixes**
  - Removed outdated case risk summary details from the display.
- **Chores**
  - Updated mock data to reflect new company name and tagging for workflow management.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alonp99 committed:
refactor(overview): improve case overview tab handling and labels

- Update tab labels for clarity and consistency
- Simplify tab entry lookup by using Object.entries
- Enhance risk indicators mapping for improved readability

tomer-shvadron commented on apps/backoffice-v2/src/pages/Entity/components/Case/components/CaseOverview/CaseOverview.tsx:
```diff
@@ -29,25 +29,29 @@ export const CaseOverview = ({ processes }: { processes: string[] }) => {
+      const domainTitle = domain ?? '';
+      const tabEntry = Object.entries(TabToLabel).find(([_, label]) => label === domainTitle);
+      const tab = tabEntry ? tabEntry[0] : camelCase(domainTitle.toLowerCase());
+      const isValidCaseTab = CaseTabs.includes(tab as any);
```
Let's not add `any` type
```suggestion
      const isValidCaseTab = CaseTabs.includes(tab as keyof typeof CaseTabs);
```

tomer-shvadron commented on apps/backoffice-v2/src/pages/Entity/components/Case/components/CaseOverview/CaseOverview.tsx:
```diff
@@ -29,25 +29,29 @@ export const CaseOverview = ({ processes }: { processes: string[] }) => {
-    };
-  });
+      return {
+        title: domain,
```
Should this be `domainTitle`?
```suggestion
        title: domainTitle,
```

alonp99 committed:
refactor(components): clean up AISummaryBlock and improve layout

- Remove redundant sections and clean up formatting in AISummaryBlock
- Rename 'Monitoring Reports' tab to 'Web Presence' for clarity
- Update demo company names for consistency across components

alonp99 committed:
Update apps/backoffice-v2/src/pages/Entity/components/Case/components/CaseOverview/CaseOverview.tsx

Co-authored-by: Tomer Shvadron <tomers@ballerine.com>

alonp99 committed:
Merge branch 'dev' into BAL-3725
